crypto/rand/drbg_lib.c

```diff
@@ -472,16 +558,51 @@ int RAND_DRBG_set_callbacks(RAND_DRBG *drbg,
 
 /*
  * Set the reseed interval.
+ *
+ * The drbg will reseed automatically whenever the number of generate
+ * requests exceeds the given reseed interval.
+ * Returns 1 on success, 0 on failure.
  */
-int RAND_DRBG_set_reseed_interval(RAND_DRBG *drbg, int interval)
+int RAND_DRBG_set_reseed_interval(RAND_DRBG *drbg, unsigned int interval)
 {
-    if (interval < 0 || interval > MAX_RESEED)
+    if (interval > MAX_RESEED_INTERVAL)
         return 0;
     drbg->reseed_interval = interval;
     return 1;
 }
 
 /*
+ * Enable the reseed timer (and set the reseed time interval)
+ *
+ * The reseed timer works differently depending on whether the DRBG
+ * is free (drbg->parent == NULL) or chained (drbg->parent != NULL):
+ *
+ * For a free DRBG, the |interval| argument specifies the reseed_time_interval,
+ * i.e., the number of seconds after which the DRBG shall reseed automatically.
+ * Specifying a reseed_time_interval of 0 disables the reseed timer.
+ *
+ * For a child DRBG, the |interval| argument is just used to enable or disable
+ * the reseed timer, depending on whether its value is nonzero or not.
+ * If the reseed timer is enabled, the DRBG will reseed itself whenever its
+ * reseed_time differs from the parent's reseed_time.
```

Comment:
I guess that "reseed timer" is confusing, since there actually is no timer involved.
It's a way for a child DRBG to reseed when the parent DRBG reseeded.
I'm wondering if we should split the functionality to make it more clear what it does.

Reply:
You are right. I wanted to replace "reseed timer" by "automatic reseeding", but forgot to do it:

```
 * For a child DRBG, the |interval| argument is just used to enable or disable
 * automatic reseeding, depending on whether its value is nonzero or not.
 * If automatic reseeding is enabled, the DRBG will reseed itself whenever its
 * reseed_time differs from the parent's reseed_time.
```

> I'm wondering if we should split the functionality to make it more clear what it does.

I was pondering about that variant, too. But I abandoned it, because the current implementation appeared simpler to me. I think the implementation itself is ok, only the explanation is not clear enough. So I will try to improve the comments.